crates/pipeline-types/src/transport/datagen.rs

```diff
@@ -0,0 +1,169 @@
+use crate::config::TransportConfigVariant;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use std::collections::HashMap;
+use std::num::NonZeroU32;
+use utoipa::ToSchema;
+
+fn default_scale() -> i64 {
+    1
+}
+
+fn default_exponent() -> usize {
+    1
+}
+
+fn default_workers() -> usize {
+    1
+}
+
+fn default_sequence() -> Vec<GenerationPlan> {
+    vec![GenerationPlan::default()]
+}
+
+/// Describe the strategy used to generate values.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+#[serde(tag = "name")]
+pub enum Strategy {
```

Comment:
If we plan to expose this type via OpenAPI, it should probably have a qualified name like `DatagenStrategy`.